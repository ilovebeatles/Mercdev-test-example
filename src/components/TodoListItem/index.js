import React from 'react';
import styled from 'styled-components';
import { observer } from 'mobx-react';

const TodoItem = styled.div`
    ${props => props.done
        ? `color: red; text-decoration: line-through;`
        : `color: #61DAFB;`
    }
`;

class TodoListItem extends React.Component {
  onClickClose = () => {
    const index = parseInt(this.props.index);
    this.props.removeItem(index);
  }

  onChange = () => {
    const index = parseInt(this.props.index);
    this.props.markTodoDone(index);
  }

  render() {
    return(
      <li className="list-group-item ">
        <TodoItem done={this.props.item.done}>
          <label>
            <input type="checkbox" checked={this.props.item.done} onChange={this.onChange} />
            {this.props.item.value}
          </label>
          <button type="button" className="close" onClick={this.onClickClose}>&times;</button>
        </TodoItem>
      </li>
    );
  }
}

export default observer(TodoListItem);
