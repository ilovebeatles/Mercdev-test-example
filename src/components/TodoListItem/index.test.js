import React from 'react';
import { shallow } from 'enzyme'
import TodoListItem from './';

it('renders correctly', () => {
  const item = {
    done: true,
    value: 'test'
  };
  const component = shallow(<TodoListItem  item={item}/>);
  expect(component).toMatchSnapshot();
});
