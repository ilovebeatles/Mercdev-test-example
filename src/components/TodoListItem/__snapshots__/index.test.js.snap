// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoListItem
    item={
      Object {
        "done": true,
        "value": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ForwardRef
        done={true}
      >
        <label>
          <input
            checked={true}
            onChange={[Function]}
            type="checkbox"
          />
          test
        </label>
        <button
          className="close"
          onClick={[Function]}
          type="button"
        >
          ×
        </button>
      </ForwardRef>,
      "className": "list-group-item ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <label>
            <input
              checked={true}
              onChange={[Function]}
              type="checkbox"
            />
            test
          </label>,
          <button
            className="close"
            onClick={[Function]}
            type="button"
          >
            ×
          </button>,
        ],
        "done": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={true}
                onChange={[Function]}
                type="checkbox"
              />,
              "test",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": true,
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "test",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "×",
            "className": "close",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "×",
          "type": "button",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
    ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ForwardRef
          done={true}
        >
          <label>
            <input
              checked={true}
              onChange={[Function]}
              type="checkbox"
            />
            test
          </label>
          <button
            className="close"
            onClick={[Function]}
            type="button"
          >
            ×
          </button>
        </ForwardRef>,
        "className": "list-group-item ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <label>
              <input
                checked={true}
                onChange={[Function]}
                type="checkbox"
              />
              test
            </label>,
            <button
              className="close"
              onClick={[Function]}
              type="button"
            >
              ×
            </button>,
          ],
          "done": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={true}
                  onChange={[Function]}
                  type="checkbox"
                />,
                "test",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": true,
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "test",
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "×",
              "className": "close",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "×",
            "type": "button",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "rules": Array [
              "
    ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
